airbyte-integrations/connectors/source-iterable/source_iterable/api.py

```diff
@@ -158,6 +175,64 @@ def path(self, **kwargs) -> str:
         return "campaigns"
 
 
+class CampaignsMetrics(IterableStream):
+    primary_key = None
+    data_field = None
+
+    def __init__(self, api_key: str, start_date: str):
+        """
+        https://api.iterable.com/api/docs#campaigns_metrics
+        """
+        super().__init__(api_key)
+        self.start_date = start_date
+
+    def path(self, **kwargs) -> str:
+        return "campaigns/metrics"
+
+    def request_params(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> MutableMapping[str, Any]:
+        params = super().request_params(**kwargs)
+        params["campaignId"] = stream_slice.get("campaign_id")
+        params["startDateTime"] = self.start_date
+
+        return params
+
+    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+        lists = Campaigns(api_key=self._api_key)
+        for list_record in lists.read_records(sync_mode=kwargs.get("sync_mode", SyncMode.full_refresh)):
+            yield {"campaign_id": list_record["id"]}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        content = response.content.decode()
+        yield {"data": self._parse_csv_string_to_dict(content)}
+
+    @staticmethod
+    def _parse_csv_string_to_dict(csv_string: str) -> Dict[str, Any]:
```

Comment:
important to note that we assume that there is only one line in csv input

Reply:
mb move it outside of the class and make it return list of objects?

Reply:
Technically there are two lines, the first one is for variable names, and the second – for values.
This is used only here at the moment, and we can extract it when necessary.
We return CampaignMetircs as a raw json, so converting to object and then back to dict looks unnecessary, what do you think?